### PR TITLE
Dockerizing neurodsp

### PR DESCRIPTION
**Pull request by GrantRVD**
This pull requests adds a `Dockerfile` that fully dockerizes the repository and tutorials, allowing for execution of ipython notebooks or development of the repository in a container environment. It includes a `Makefile` to facilitate launching and managing the container for those who may not be familiar with Docker's CLI. The Docker image itself uses the official `continuumio/miniconda3` image as a base and, when fully running, uses less than 2GB, as opposed to 4.5GB for the full anaconda3 image. The image is configured to serve over port 5555 by default, leaving Jupyter's usual port 8888 open.

I found that my machine was generating some extra build and egg-info files in development, so I also updated the `.gitignore` to include the standard Python repository ignore-files, and I appended the custom ignores from the previous version.

I had a lot of fun making this, since I only learned Docker recently, and I hope that this can serve as a preliminary template for how this and future Voytek Lab repositories can use Docker to enhance cross-platform usability.

`CONTRIBUTE.md` doesn't specify how to update the README, but I would be happy to make those additions if the requirements are clarified.

footnote: the `filters` notebook had a runtime error that I'm currently debugging and might submit an issue for later, but I decided this was enough for now. The `Phase Estimation by Extrema Interpolation` notebook also has an error, but this is present on the repository version so I haven't considered this a new error.

### CODE REVIEW

**Comment by GrantRVD** [comment]
Not sure what the error is with Travis CI, as none of the codebase was modified in this PR. I'm not aware of any required additions to the YAML to make it work, at least.

**Comment by srcole** [comment]
Thanks for the contribution! I'm not too familiar with Docker, but what you made sounds/looks reasonable, superficially. I'm only worried putting it into the repo because it breaks Travis CI, and I like to keep that functional. If anyone figures out how to make Travis CI play nice with the dockerfile, I'm happy to merge the PR.

Speaking of Docker, I'll also set up neurodsp with [binder](https://mybinder.org/).

And thanks for mentioning those 2 errors in the tutorial notebooks. Will make a PR to fix those shortly.

**Comment by GrantRVD** [comment]
It looks like the problem might have been on the TravisCI end of things not being able to pull the needed versions of Python at the time of submission. Right now, it looks like versions 3.4 and 3.6 tests pass but not 3.5, weirdly enough. I'm not sure if you can manually re-start the Travis test but it looks like the problem might fix itself eventually.

Right now, the error for TravisCI looks like this
```
W: The repository 'http://ppa.launchpad.net/couchdb/stable/ubuntu trusty Release' does not have a Release file.
W: Failed to fetch http://ppa.launchpad.net/git-core/ppa/ubuntu/dists/trusty/InRelease  Could not connect to ppa.launchpad.net:80 (91.189.95.83), connection timed out
W: Failed to fetch http://ppa.launchpad.net/openjdk-r/ppa/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/pollinate/ppa/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/webupd8team/java/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/chris-lea/redis-server/ubuntu/dists/trusty/main/binary-amd64/Packages  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/chris-lea/redis-server/ubuntu/dists/trusty/main/binary-i386/Packages  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/chris-lea/redis-server/ubuntu/dists/trusty/main/i18n/Translation-en  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/main/binary-amd64/Packages  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/main/binary-i386/Packages  Unable to connect to ppa.launchpad.net:http:
W: Some index files failed to download. They have been ignored, or old ones used instead.

127.0.0.1 localhost nettuno travis vagrant
127.0.1.1 travis-job-efa1e3aa-8aab-4f68-a8f6-34ef69f6d6ba travis-job-efa1e3aa-8aab-4f68-a8f6-34ef69f6d6ba ip4-loopback trusty64

[33;1m3.5 is not installed; attempting download[0m
[33;1mDownloading archive: https://s3.amazonaws.com/travis-python-archives/binaries/ubuntu/14.04/x86_64/python-3.5.tar.bz2[0m
curl: (35) Unknown SSL protocol error in connection to s3.amazonaws.com:443 

[31;1mThe command "curl -s -o python-3.5.tar.bz2 ${archive_url}" failed and exited with 35 during .[0m

Your build has been stopped.
```
Long story short, it looks like no further changes to the `travis.yml` are needed.

**Comment by GrantRVD** [comment]
Now it says version 3.5 passed and 3.4 had the problem. This is fun.

I updated a reference to the old version of this module (`braindsp`) in the YAML file just so I could submit another commit and hopefully it'll pass all the Python installations this time.

**Comment by srcole** [comment]
looks like they all passed this time!
